### PR TITLE
Try to fix the .git/config file is not changed when git url is changed on configuration

### PR DESCRIPTION
**Pull request by zhanglong2015**
When we have the case as the following step 

1. create a job and input git url , then the job runs and works well
2. For some reason, we need change the git url in job configuration. The build works but the remote url in .git/config file is not changed. 
    Actually , we need read build manifest like remote url from .git/confg file, since the url is not changed. we got wrong manifest. 
    My proposal is to check if the remote url is changed from job configuration once change, wipe out work space in checkout method.

### CODE REVIEW

**Comment by Vlatombe** [comment]
I don't see the point of wiping the workspace, only the git config should be affected.
You should check out the current implementation in GitSCM#fetchFrom and understand what is going wrong there (from my understanding that should already do what you described)

**Comment by MarkEWaite** [comment]
There appear to be no tests to verify the old behavior, and no tests added in this commit.

Without tests, there is a high risk that the feature you felt was important to add will be broken by a future change, and the break will not be detected.  Please add tests to show that you can detect the problem (without your changes), then apply your changes and confirm that the problem is fixed.

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 1]
```diff
@@ -1598,4 +1642,4 @@ public static void configureXtream() {
      * To avoid pointlessly large changelog, we'll limit the number of changes up to this.
      */
     public static final int MAX_CHANGELOG = Integer.getInteger(GitSCM.class.getName()+".maxChangelog",1024);
-}
+}
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: ?

**Comment by zhanglong2015** [comment]
Yes, you are right. After using 2.2.5 ,the problem is gone. Please close this request , Thanks you !

**Pull request closed without merging**